init.lua

```diff
@@ -510,56 +496,66 @@ end
 
 local function get_filter_items(data, player)
 	local filter = data.filter
-	if searches[filter] then
-		data.items = searches[filter]
-		return
+	local items_list
+	if filter == "" then
+		items_list = init_items
+	elseif searches[filter] then
+		items_list = searches[filter]
+	else
+		local c = 0
+		items_list = {}
+		for i = 1, #init_items do
+			local item = init_items[i]
+			local item_desc = reg_items[item].description:lower()
+			if item:find(filter, 1, true) or item_desc:find(filter, 1, true) then
+				c = c + 1
+				items_list[c] = item
+			end
+		end
+		searches[filter] = items_list
```

Comment:
Here you’re changing the current caching system. You cache the items after a single search whereas I wanted two searches (to avoid caching on silly searches).

Reply:
I chache only the full-list => pattern-search result that should be constant in a session for all users. The second part, the progressive filter, is not cached and done each call

Reply:
Yes, but you cache it after only one search. That’s the problem for me.